agent/consul/leader_peering_test.go

```diff
@@ -615,7 +616,7 @@ func insertTestPeeringData(t *testing.T, store *state.Store, peer string, lastId
 	return lastIdx
 }
 
-// TODO(peering): once we move away from leader only request for PeeringList, move this test to consul/server_test maybe
+// TODO(peering): once we move away from keeping state in stream tracker only on leaders, move this test to consul/server_test maybe
```

Comment:
the wording was bugging me here; ( i wrote it hehe 😅 )

It's not the PeeringList endpoint's problem, but rather the fact that we rely on the stream tracker information that requires us to make these `leader` only for now